sdk/spring/spring-integration-azure-eventhubs/src/main/java/com/azure/spring/integration/eventhubs/inbound/EventHubInboundChannelAdapter.java

```diff
@@ -41,9 +46,11 @@
     private final String consumerGroup;
     private final ListenerMode listenerMode;
     private final IntegrationRecordEventProcessingListener recordEventProcessor = new IntegrationRecordEventProcessingListener();
+    private final IntegrationBatchEventProcessingListener batchEventProcessor =
+        new IntegrationBatchEventProcessingListener();
     private final CheckpointConfig checkpointConfig;
     private EventProcessingListener listener;
-    private CheckpointManager checkpointManager;
+    private EventCheckpointManager checkpointManager;
```

Comment:
I think we should use an interface here instead of type, using `EventCheckpointManager` this type will make it impossible for users to provide their own implementation.

Reply:
If we are going to allow users to bring their own implementation, we should move this field to the EventProcessingListener and expose a set method just as we did for the message converter.

Reply:
EventCheckpointManager is returned from CheckpointManagers.of(CheckpointConfig checkpointConfig, ListenerMode listenerMode) method, and this is not to be extended by user for now. the behavior here is hard coded in the if-else block, use implementation is not supported I think.

Reply:
ok, let's keep it this way then.